### PR TITLE
Note sharding persistent mode deprecation in migration guide

### PR DESCRIPTION
**Pull request by chbatey**
Refs #26177

### CODE REVIEW

File: akka-docs/src/main/paradox/project/migration-guide-2.5.x-2.6.x.md [hunk 1]
```diff
@@ -199,6 +199,12 @@ Akka is now using Protobuf version 3.9.0 for serialization of messages defined b
 Cluster client has been deprecated as of 2.6 in favor of [Akka gRPC](https://doc.akka.io/docs/akka-grpc/current/index.html).
 It is not advised to build new applications with Cluster client, and existing users @ref[should migrate to Akka gRPC](../cluster-client.md#migration-to-akka-grpc).
 
+### Persistent mode for Cluster Sharding
+
+Cluster Sharding coordinator and Replicated Entities state in 2.5 can be stored in Distributed Data or via Akka Persistence.
```
Comment (patriknw) [inline akka-docs/src/main/paradox/project/migration-guide-2.5.x-2.6.x.md]: it's not Replicated Entities, but "Remembering Entities"
maybe make that a link as well typed/cluster-sharding.md#remembering-entities

**Review by patriknw — APPROVED** [review]
LGTM, after minor nit

**Review by johanandren — APPROVED** [review]

**Review by raboof — APPROVED** [review]
Updated the link

**Comment by patriknw** [comment]
thanks

**Pull request merged**